src/main/java/com/zontik/groshiky/service/TransactionService.java

```diff
@@ -48,26 +49,25 @@ public void deleteTransactionById(Integer id) {
 
     @Override
     public Transaction findTransactionById(Integer id) {
-        return transactionRepository.getOne(id);
+        return transactionRepository.findById(id).orElse(null);
     }
 
     @Override
-    public Transaction editTransaction(Integer account_id, Integer id, Transaction transaction) {
-        Account account = (accountRepository.findAllById(account_id));
-        Transaction tr = transactionRepository.getOne(id);
-        tr.setId(id);
-        tr.setTransactionType(transaction.getTransactionType());
-        tr.setName(transaction.getName());
-        tr.setDate(transaction.getDate());
+    public Transaction editTransaction(TransactionDto transactionDto) {
+        Account account = (accountRepository.findAllById(transactionDto.getAccountId()));
+        Transaction tr = transactionRepository.findById(transactionDto.getId()).orElse(null);
         Double newBalance = (account.getBalance())-tr.getAmount();
-        if(transaction.getTransactionType() == TransactionType.INCOME){
-            account.setBalance(newBalance + transaction.getAmount());
+        if(transactionDto.getTransactionType().equals(TransactionType.INCOME)){
+            account.setBalance(newBalance + transactionDto.getAmount());
         }else{
-            transaction.setAmount(transaction.getAmount() * (-1));
-            account.setBalance(newBalance + transaction.getAmount());
+            transactionDto.setAmount(transactionDto.getAmount() * (-1));
+            account.setBalance(newBalance + transactionDto.getAmount());
         }
-        tr.setAmount(transaction.getAmount());
+        tr.setAmount(transactionDto.getAmount());
         tr.setAccount(account);
+        tr.setId(transactionDto.getId());
+        tr.setTransactionType(transactionDto.getTransactionType());
+        tr.setName(transactionDto.getName());
         return transactionRepository.save(tr);
```

Comment:
all this method should be replaced by only one call of modelMapper: 
modelMapper.map(dto, entity);  :)